### ISSUE TITLE
Attach pre/post methods to RPC endpoints

### ISSUE THREAD

**Issue by brahmlower** [bplower/cssef#49]
An issue with the RPC model right now is that there's no way for the client to trigger pre/post work on data being sent to or received from remote endpoints. A great example of this is exemplified by the commit message in b1a697eb52becae4e2139af2b017da5ac9476da4 where the result of calling 'login' is that the auth token is printed to the screen rather than saved in the auth token file. The token should never been printed to the screen, and should instead be written to the token file.

In short: add client support for pre/post method triggers to RPC calls.
